store/db/goleveldb.go

```diff
@@ -0,0 +1,426 @@
+package db
+
+import (
+	"bytes"
+	"fmt"
+	"path/filepath"
+
+	"github.com/spf13/cast"
+	"github.com/syndtr/goleveldb/leveldb"
+	"github.com/syndtr/goleveldb/leveldb/errors"
+	"github.com/syndtr/goleveldb/leveldb/filter"
+	"github.com/syndtr/goleveldb/leveldb/iterator"
+	"github.com/syndtr/goleveldb/leveldb/opt"
+	"github.com/syndtr/goleveldb/leveldb/util"
+
+	corestore "cosmossdk.io/core/store"
+)
+
+// GoLevelDB implements RawDB using github.com/syndtr/goleveldb/leveldb.
+// It is used for only store v2 migrations, since iavl v0/v1 used goleveldb.
```

Comment:
we can't guarantee this though. Some chains don't use goleveldb, they could use rocks.